src/port1.0/portbump.tcl

```diff
@@ -0,0 +1,316 @@
+            }
+
+            set missing_types $portchecksum::default_checksum_types
+
+            # Append the string for the calculated types and note any of
+            # our default types that were already calculated
+            if {[info exists calculated_checksums_array($distfile)] && [llength $calculated_checksums_array($distfile)]} {
+                set calculated_checksums $calculated_checksums_array($distfile)
+                foreach {type sum} $calculated_checksums {
+                    set found [lsearch -exact ${missing_types} ${type}];
+                    if { ${found} != -1} {
+                        set missing_types [lreplace ${missing_types} ${found} ${found}]
+                    }
+                }
+            }
+
+            # Append the string for any of our default types that were
+            # not previously calculated
+            if {[llength $missing_types]} {
+                # get the full path of the distfile.
+                set fullpath [file join $distpath $distfile]
+                if {![file isfile $fullpath]} {
+                    return -code error "$distfile does not exist in $distpath"
+                }
+                # foreach type $missing_types {
+                #     lappend both_checksums $type "-" [portchecksum::calc_$type $fullpath]
+                # }
+            }
+        }
+
+        global version revision
+
+        ui_msg "We will bump these:"
+        foreach {type sum calculated_sum} $both_checksums {
+            ui_msg [format "Old %-8s %s" ${type}: $sum]
+            ui_msg [format "New %-8s %s" ${type}: $calculated_sum]
+        }
+
+        if {[info exists macports::ui_options(questions_yesno)]} {
+            set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 "Do you approve of these changes?"]
+            if {$retval == 0} {
+                # User said yes
+                # TODO: Ask to reset revision number only if > 0 and pass revision number to message.
+                set retval [$macports::ui_options(questions_yesno) $msg "BumpPrompt" "" {y} 0 [format "Do you want to reset revision number (currently %s) to 0?" $revision]]
+                if {$retval == 0} {
+                    # User said yes
+                    # TODO: Reset revision to 0
+                }
+            } else {
+                # User said no
+                return 0
+            }
+        }
+
+        set patterns {}
+
+        set whitespace {[[:space:]\\]+}
+        # Create substitution pattern(s) for checksum
+        foreach {type sum calculated_sum} $both_checksums {
+            lappend patterns "s/(${type}${whitespace})${sum}/\\1${calculated_sum}/g"
+        }
+        # Create substitution pattern for revision (reset to 0)
+        lappend patterns {s/(revision[[:space:]\\]+)[0-9]+/\10/g}
+
+        # Construct sed command
+        set cmdline {}
+        lappend cmdline $portutil::autoconf::sed_command -E
+        foreach {pattern} $patterns {
```

Comment:
Why the quotes around `pattern` here? Those don't seem to be required to me?

Reply:
My mistake, I copied the foreach from earlier lines. Removed